app/Models/Sink.php

```diff
@@ -56,4 +56,11 @@ public function isLive(): Attribute
     {
         return Attribute::make(get: fn(mixed $value, array $attr) => $attr['status'] === 'live');
     }
+
+    public function hasDoc(): Attribute
+    {
+        return Attribute::make(
+            get: fn() => (new ($this->impl_class)())->getDocumentation() !== null
+        );
+    }
 }
```

Comment:
I'm fine with this. It's a bit overhead, but the number of sinks will always be small, så all good. Neat solution!